src/components/sequencing/actions/ActionRunCard.svelte

```diff
@@ -0,0 +1,89 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import PlayBtnIcon from 'bootstrap-icons/icons/play-btn.svg?component';
+  import StopwatchIcon from 'bootstrap-icons/icons/stopwatch.svg?component';
+  import { Status } from '../../../enums/status';
+  import type { ActionDefinition, ActionRun, ActionRunSlim } from '../../../types/actions';
+  import { convertUsToDurationString } from '../../../utilities/time';
+  import StatusBadge from '../../ui/StatusBadge.svelte';
+
+  export let actionRun: ActionRunSlim;
+  export let actionDefinition: ActionDefinition | null;
+  export let interactable: boolean = true;
+
+  function getStatusForActionRun(actionStatus: ActionRun['status']): Status {
+    if (actionRun.error || actionRun.results?.status === 'FAILED') {
+      return Status.Failed;
+    }
+
+    switch (actionStatus) {
+      case 'complete':
+        return Status.Complete;
+      case 'pending':
+        return Status.Pending;
+      case 'in-progress':
+        return Status.Incomplete;
+      case 'failed':
+        return Status.Failed;
+      default:
+        return Status.Unchecked;
+    }
+  }
+
+  function formatDuration(duration: number | null): string {
```

Comment:
Maybe move this outside to a helper file so we can use it to format other durations?

Reply:
Good idea